### PR TITLE
drivers:ad7124: drivers enhancement

### PR DESCRIPTION
**Pull request by D-Disha**
Added Device Configurations.
Made editorial changes like function return types
and removal of brackets for single line if-else comments

### CODE REVIEW

File: drivers/adc/ad7124/ad7124.c [hunk 1]
```diff
@@ -761,35 +677,419 @@ int32_t ad7124_set_odr(struct ad7124_dev *dev, float odr,
...
+	dev->active_device = init_param->active_device;
+
+	/* Read ID register to identify the part. */
+	ret = ad7124_read_register(dev, &dev->regs[AD7124_ID_REG]);
+	if (ret)
+		goto error_spi;
+	if (dev->active_device == ID_AD7124_4) {
+		if (!(dev->regs[AD7124_ID_REG].value = AD7124_4_ID))
+			goto error_spi;
+	} else if (dev->active_device == ID_AD7124_8) {
```
Comment (jsanbuen) [inline drivers/adc/ad7124/ad7124.c:1016]: This can be changed to else from else if since there seems to be only two conditions that need to be observed: ID matches AD7124_ID or not.

File: drivers/adc/ad7124/ad7124.h [hunk 2]
```diff
@@ -257,19 +265,139 @@
...
+	AD7124_SYS_ZERO_SCALE_OFF,
+	AD7124_SYS_ZERO_SCALE_GAIN,
+	ADC_MAX_MODES
+};
+
+/**
+ * @enum ad7124_analog_input
+ * @brief ADC input sources for each channel.
+ **/
+enum ad7124_analog_input {
```
Comment (jsanbuen) [inline drivers/adc/ad7124/ad7124.h:312]: No need to explicitly declare the value of each enum. Only declare explicit value for enum member that would be breaking the sequence i.e., if next enum value from 0x06 would be 0x10.

File: drivers/adc/ad7124/ad7124.h [hunk 3]
```diff
@@ -257,19 +265,139 @@
...
+	bool channel_enable;
+	uint8_t setup_sel;
+	struct ad7124_analog_inputs ain;
+};
+
+/**
+ * @enum ad7124_reference_source
+ * @brief Type of ADC Reference
+**/
+enum ad7124_reference_source {
```
Comment (jsanbuen) [inline drivers/adc/ad7124/ad7124.h:368]: Same comment on explicit enum value declaration

File: drivers/adc/ad7124/ad7124.h [hunk 4]
```diff
@@ -257,19 +265,139 @@
...
 	int32_t addr;
 	int32_t value;
 	int32_t size;
 	int32_t rw;
 };
 
-/*! AD7124 registers list*/
+/* AD7124 registers list */
 enum ad7124_registers {
 	AD7124_Status = 0x00,
```
Comment (jsanbuen) [inline drivers/adc/ad7124/ad7124.h:402]: No need to add 0x00 explicit declaration. First member will have value of 0 already.

File: drivers/adc/ad7124/ad7124.c [hunk 5]
```diff
@@ -92,110 +85,98 @@ int32_t ad7124_no_check_read_register(struct ad7124_dev *dev,
...
 			     AD7124_COMM_REG_RA(p_reg->addr);
-		for(i = 1; i < p_reg->size + 2 + add_status_length; ++i) {
+		for (i = 1; i < p_reg->size + 2 + add_status_length; ++i)
 			msg_buf[i] = buffer[i];
-		}
 		check8 = ad7124_compute_crc8(msg_buf, p_reg->size + 2 + add_status_length);
 	}
 
-	if(check8 != 0) {
+	if (check8 != 0)
```
[the remark this answers is not in the document]
Comment (D-Disha) [reply drivers/adc/ad7124/ad7124.c:108]: Will change that

File: drivers/adc/ad7124/ad7124.c (hunk 1 above)
Comment (D-Disha) [reply drivers/adc/ad7124/ad7124.c:1016]: Yes will change that

File: drivers/adc/ad7124/ad7124.h (hunk 2 above)
Comment (D-Disha) [reply drivers/adc/ad7124/ad7124.h:312]: Will change that.

**Comment by buha** [comment]
i'm approving as-is but please, in the future, separate editorial changes from functional ones like adding functions, adding new types/enums etc.

**Review by buha — APPROVED** [review]

**Review by amiclaus — CHANGES_REQUESTED** [review]
please split the changes into 2 commits:
- non-functional changes (indentations/comments/moving block codes, etc)
- functional changes (new implementations, added features)

We want to keep the commit history clean.

Also, each commit should end with `Signed-off by:`

Comments like "addesssed reviewer changes" should not appear in the commit body.

**Review by jsanbuen — APPROVED** [review]
Approving the changes made but please follow the other reviewers' guidelines on the commits. Double check if the changes implemented are just syntax / editorial change or includes functional changes.

**Review by amiclaus — CHANGES_REQUESTED** [review]
please remove the first 2 commits (the initial commit 16cc873e2e21be6c8150fa681bbb45c5a6d93994 which is now duplicated and the merge commit since we only do rebase and merge on this repo).

Also, breifly looking at your new commits (b4fc7fe6b25bebdfad3151671d0982fdca03acaf) in one of them you remove a set of functions and in the next one you add them again. As I mentioned before: please add a commit only with non-functional changes and one with functional changes, at least. I'd suggest using `git rebase --interactive` to handle your edits with the commits.

File: drivers/adc/ad7124/ad7124.h [hunk 6]
```diff
@@ -351,101 +471,163 @@ struct ad7124_dev {
...
+/* Initializes the AD7124 */
 int32_t ad7124_setup(struct ad7124_dev **device,
 		     struct ad7124_init_param *init_param);
-/*! Free the resources allocated by AD7124_Setup(). */
+
+/* Free the resources allocated by ad7124_setup(). */
 int32_t ad7124_remove(struct ad7124_dev *dev);
 
-#endif /* __AD7124_H__ */
+#endif /* __AD7124_H__ */
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.h:633]: nit: new line ad the end of the file

**Comment by D-Disha** [comment]
Please note that the power mode has been added to device structure and some minor changes in ad7124_fltcoeff_get() and ad7124_fclk_get() since last review.

**Review by jsanbuen — APPROVED** [review]

File: drivers/adc/ad7124/ad7124.c [hunk 7]
```diff
@@ -92,110 +85,98 @@ int32_t ad7124_no_check_read_register(struct ad7124_dev *dev,
...
 			     AD7124_COMM_REG_RA(p_reg->addr);
-		for(i = 1; i < p_reg->size + 2 + add_status_length; ++i) {
+		for (i = 1; i < p_reg->size + 2 + add_status_length; ++i)
 			msg_buf[i] = buffer[i];
-		}
 		check8 = ad7124_compute_crc8(msg_buf, p_reg->size + 2 + add_status_length);
 	}
 
-	if(check8 != 0) {
+	if (!check8)
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.c:108]: this looks like a functional change. Why `if (!check8)` instead of `if (check8)`

File: drivers/adc/ad7124/ad7124.c [hunk 8]
```diff
@@ -439,170 +417,105 @@ int32_t ad7124_read_data(struct ad7124_dev *dev,
...
 	uint8_t crc = 0;
 
-	while(buf_size) {
-		for(i = 0x80; i != 0; i >>= 1) {
+	while (buf_size) {
+		for (i = 0x80; i != 0; i >>= 1) {
 			bool cmp1 = (crc & 0x80) != 0;
 			bool cmp2 = (*p_buf & i) != 0;
-			if(cmp1 != cmp2) { /* MSB of CRC register XOR input Bit from Data */
+			if (cmp1 != cmp2) {
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.c:458]: this is a general rule: if the `if` statement has brackets, then also the `else` statement should have brackets. Please revert this change.

File: drivers/adc/ad7124/ad7124.h (hunk 6 above)
Comment (amiclaus) [reply drivers/adc/ad7124/ad7124.h:647]: issue still persists.

File: drivers/adc/ad7124/ad7124.c [hunk 9]
```diff
@@ -760,4 +1092,4 @@ int32_t ad7124_remove(struct ad7124_dev *dev)
 	no_os_free(dev);
 
 	return 0;
-}
+}
```
Comment (amiclaus) [inline drivers/adc/ad7124/ad7124.c:1095]: nit: new line missing.

File: drivers/adc/ad7124/ad7124.c (hunk 7 above)
Comment (D-Disha) [reply drivers/adc/ad7124/ad7124.c:108]: This is one of the first comments I got in this PR as suggested by jsanbuen, I have changed it because if(check8 !=0)  is same as checking if (!check8).  Hope that makes sense.

Comment (amiclaus) [reply drivers/adc/ad7124/ad7124.c:108]: What I am trying to understand: which case should return an error code? `if (!check8)` or `if (check8)`? Because these 2 statements are completely different.

Comment (D-Disha) [reply drivers/adc/ad7124/ad7124.c:108]: It should actually be if(check8), will change it.

**Review by amiclaus — APPROVED** [review]

**Review by amiclaus — CHANGES_REQUESTED** [review]
it seems that a merge commit was added in the meantime. please remove it.

**Review by amiclaus — APPROVED** [review]

**Review by jsanbuen — APPROVED** [review]

**Review by jsanbuen — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]
